File: NotificationSystem/NotificationSystem.Web/ctrAdmin.ascx.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.UI;
using System.Web.UI.WebControls;
using NotificationSystem.NotificationSystem.Data.NotificationSystemTableAdapters;
using static NotificationSystem.NotificationSystem.Data.NotificationSystem;
using NotificationSystem.NotificationSystem.Data.Classes;
using Microsoft.VisualBasic;
using System.Data;

namespace NotificationSystem.NotificationSystem.Web
{
	public partial class ctrAdmin : System.Web.UI.UserControl
	{
        private string m_UserName = string.Empty;
        private string m_Password = string.Empty;

        public string UserName
        {
            get
            {
                return m_UserName;
            }
            set
            {
                m_UserName = value;
            }
        }
             public string Password
        {
            get
            {
                return m_Password;
            }
            set
            {
                m_Password = value;
            }
        }


        protected void Page_Load(object sender, EventArgs e)
		{
            clsNotificationSystem theNotificationSystem = new clsNotificationSystem();
            AdminDataTable tblGetAdmin = new AdminDataTable();


            try
            {

                if (Request.Form["ctl00$MainContent$ctrAdmin_Search$btnSearch"] == "Search")
                    ctrHiddebField.Value = Convert.ToString(m_UserName);


                if ((Page.IsPostBack) & this.lblSearchResult.Text.Length > 0)
                    tblGetAdmin = (AdminDataTable)theNotificationSystem.GetAdmin(m_UserName);

                this.lblSearchResult.Text = tblGetAdmin.Rows.Count + " Result(s)";
                this.grdAdmin.DataSource = tblGetAdmin.DefaultView;
                this.grdAdmin.DataBind();
            }

            catch (Exception)
            {
                throw;
            }
        }

        private void grdAdmin_PageIndexChanging(object sender, GridViewPageEventArgs e)
        {
            try
            {
                if (!(Information.IsNothing(ViewState["columnname"]) | Information.IsNothing(ViewState["direction"])))
                {
                    DataView m_DataView = (DataView)this.grdAdmin.DataSource;

                    if (m_DataView == null)
                    {
                        m_DataView.Sort = ViewState["columnname"].ToString() + " " + ViewState["direction"].ToString();
                        this.grdAdmin.DataSource = m_DataView;
                    }
                }

                this.grdAdmin.PageIndex = e.NewPageIndex;
                this.grdAdmin.DataBind();
            }
            catch (Exception)
            {
                throw;
            }
        }

        private void grdAdmin_Sorting(object sender, GridViewSortEventArgs e)
        {
            try
            {
                DataView m_Dataview = (DataView)grdAdmin.DataSource;

                if (m_Dataview == null)
                {
                    m_Dataview.Sort = e.SortExpression + " " + ConvertSortDirection(e);
                    this.grdAdmin.DataSource = m_Dataview;
                    this.grdAdmin.DataBind();
                }
            }
            catch (Exception)
            {
                throw;
            }
        }

        private string ConvertSortDirection(System.Web.UI.WebControls.GridViewSortEventArgs e)
        {
            ViewState.Add("columnname", e.SortExpression);

            if ((ViewState["direction"] == null))
                ViewState.Add("direction", "asc");
            else
                ViewState["direction"] = Interaction.IIf(ViewState["direction"].ToString().ToLower() == "desc", "asc", "desc");

            return ViewState["direction"].ToString();
        }
    }
}

File: NotificationSystem/NotificationSystem.Web/ctrAdmin.ascx.vb
Imports System
Imports System.Web.UI
Imports System.Web.UI.WebControls
Imports Microsoft.VisualBasic
Imports System.Data

Namespace NotificationSystem.NotificationSystem.Web
    Public Partial Class ctrAdmin
        Inherits UserControl
        Private m_UserName As String = String.Empty
        Private m_Password As String = String.Empty

        Public Property UserName As String
            Get
                Return m_UserName
            End Get
            Set(value As String)
                m_UserName = value
            End Set
        End Property
        Public Property Password As String
            Get
                Return m_Password
            End Get
            Set(value As String)
                m_Password = value
            End Set
        End Property


        Protected Sub Page_Load(sender As Object, e As EventArgs)
            Dim theNotificationSystem As NotificationSystem.NotificationSystem.Data.Classes.clsNotificationSystem = New NotificationSystem.NotificationSystem.Data.Classes.clsNotificationSystem()
            Dim tblGetAdmin As NotificationSystem.NotificationSystem.Data.NotificationSystem.AdminDataTable = New NotificationSystem.NotificationSystem.Data.NotificationSystem.AdminDataTable()


            Try

                If Equals(Request.Form("ctl00$MainContent$ctrAdmin_Search$btnSearch"), "Search") Then ctrHiddebField.Value = Convert.ToString(m_UserName)


                If MyBase.Page.IsPostBack And lblSearchResult.Text.Length > 0 Then
                    tblGetAdmin = CType(theNotificationSystem.GetAdmin(m_UserName), NotificationSystem.NotificationSystem.Data.NotificationSystem.AdminDataTable)
                Else
                    tblGetAdmin = CType(theNotificationSystem.GetAdmins(), NotificationSystem.NotificationSystem.Data.NotificationSystem.AdminDataTable)
                End If

                lblSearchResult.Text = tblGetAdmin.Rows.Count.ToString() & " Result(s)"
                grdAdmin.DataSource = tblGetAdmin.DefaultView
                grdAdmin.DataBind()

            Catch __unusedException1__ As Exception
                Throw
            End Try
        End Sub





        Private Function ConvertSortDirection(e As GridViewSortEventArgs) As String
            MyBase.ViewState.Add("columnname", e.SortExpression)

            If MyBase.ViewState("direction") Is Nothing Then
                MyBase.ViewState.Add("direction", "asc")
            Else
                MyBase.ViewState("direction") = IIf(Equals(MyBase.ViewState("direction").ToString().ToLower(), "desc"), "asc", "desc")
            End If

            Return MyBase.ViewState("direction").ToString()
        End Function

        Protected Sub grdAdmin_PageIndexChanging(sender As Object, e As GridViewPageEventArgs)
            Try
                If Not (IsNothing(MyBase.ViewState("columnname")) Or IsNothing(MyBase.ViewState("direction"))) Then
                    Dim m_DataView = CType(grdAdmin.DataSource, DataView)

                    If m_DataView Is Nothing Then
                        m_DataView.Sort = MyBase.ViewState("columnname").ToString() & " " & MyBase.ViewState("direction").ToString()
                        grdAdmin.DataSource = m_DataView
                    End If
                End If

                grdAdmin.PageIndex = e.NewPageIndex
                grdAdmin.DataBind()
            Catch __unusedException1__ As Exception
                Throw
            End Try
        End Sub

        Protected Sub grdAdmin_Sorting(sender As Object, e As GridViewSortEventArgs)
            Try
                Dim m_Dataview = CType(grdAdmin.DataSource, DataView)

                If m_Dataview Is Nothing Then
                    m_Dataview.Sort = e.SortExpression & " " & ConvertSortDirection(e)
                    grdAdmin.DataSource = m_Dataview
                    grdAdmin.DataBind()
                End If
            Catch __unusedException1__ As Exception
                Throw
            End Try
        End Sub
    End Class
End Namespace
